mautic/mautic
# Refactor the Form Bundle for Symfony 3

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8001
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This refactors form bundle to work with Symfony 3.4 updates.

https://github.com/mautic/mautic/issues/8215 created, FormSubmitHelper refactoring won't be part of this PR.

https://github.com/mautic/mautic/issues/8216 created to refactor MauticFactory from SubmissionsModel. It won't be part of this PR.


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Ensure creating forms work
3. Submitting forms 
4. Mapping to core fields
